### PR TITLE
Add solver information

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: pages/ecosystem/solvers.md [hunk 1]
```diff
@@ -2,4 +2,40 @@
 
 # Solvers Ecosystem
 
-A solver is an algorithm designed to solve optimization problems.
\ No newline at end of file
+The solver ecosystem in JSO is in constant development.
+We are looking into unifying the solver structure to plug-and-play usage of our solvers, subsolvers, and tools.
+
+Within JSO, our solver development has focused on three types of packages:
+- New methods, usually acompanied by research papers
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:9]: `New methods, usually accompanied by research papers;`

File: pages/ecosystem/solvers.md [hunk 2]
```diff
@@ -2,4 +2,40 @@
...
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
+| `lbfgs`    | JSOSolvers.jl | Line search Limited memory inverse BFGS. Factorization-free. | Unconstrained            |
+| `percival` | Percival.jl   | Augmented Lagrangian. Factorization-free.                    | Any constrained          |
+| `tron`     | JSOSolvers.jl | Trust-region second-order method. Factorization-free.        | Bound-constrained        |
+| `tronls`   | JSOSolvers.jl | Least-squares versions of `tron`.                            | Unconstrained NLS        |
+| `trunk`    | JSOSolvers.jl | Trust-region second-order method. Factorization-free.        | Unconstrained            |
+| `trunkls`  | JSOSolvers.jl | Least-squares version of `trunk`.                            | Unconstrained NLS        |
+
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:27]: We can also add [RipQP.jl](https://github.com/JuliaSmoothOptimizers/RipQP.jl) for linearly-constrained quadratic programs.

File: pages/ecosystem/solvers.md [hunk 3]
```diff
@@ -2,4 +2,40 @@
...
+## Solver list
+
+Below you can see our list. We explain the types of problem below.
+
+| Solver     | Package       | Description                                                  | Types of problem         |
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
+| `lbfgs`    | JSOSolvers.jl | Line search Limited memory inverse BFGS. Factorization-free. | Unconstrained            |
+| `percival` | Percival.jl   | Augmented Lagrangian. Factorization-free.                    | Any constrained          |
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:22]: Maybe clearer than "Any constrained" we can say mixed-constrained or equality and inequality-constrained, or just general constraints?

File: pages/ecosystem/solvers.md [hunk 4]
```diff
@@ -2,4 +2,40 @@
...
+
+## Solver list
+
+Below you can see our list. We explain the types of problem below.
+
+| Solver     | Package       | Description                                                  | Types of problem         |
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
+| `lbfgs`    | JSOSolvers.jl | Line search Limited memory inverse BFGS. Factorization-free. | Unconstrained            |
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:21]: Factorization-free line search limited memory inverse BFGS.

So, we include the `factorization-free` in the "sentence", what do you think? Same for Tron, Trunk, Percival.

File: pages/ecosystem/solvers.md [hunk 5]
```diff
@@ -2,4 +2,40 @@
...
+We are looking into unifying the solver structure to plug-and-play usage of our solvers, subsolvers, and tools.
+
+Within JSO, our solver development has focused on three types of packages:
+- New methods, usually acompanied by research papers
+- Our versions of well-known methods;
+- Wrappers around noteworthy solvers.
+
+## Solver list
+
+Below you can see our list. We explain the types of problem below.
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:15]: Below you can see the list of solvers and the type of problems solved by them.

File: pages/ecosystem/solvers.md [hunk 6]
```diff
@@ -2,4 +2,40 @@
...
+
+| Solver     | Package       | Description                                                  | Types of problem         |
```
Comment (tmigot) [inline pages/ecosystem/solvers.md:17]: Personally, I found a bit strange that the titles are aligned to the right of the column.

File: pages/ecosystem/solvers.md (hunk 3 above)
Comment (tmigot) [reply pages/ecosystem/solvers.md:22]: Ok, I missed that the "any constrained" is explained below.

**Review by tmigot — COMMENTED** [review]
Thanks for the doc! I put some comments.

**Comment by abelsiqueira** [comment]
@tmigot, thanks for the review. I've addressed your comments on the new commit.

File: pages/ecosystem/solvers.md (hunk 3 above)
Comment (abelsiqueira) [reply pages/ecosystem/solvers.md]: I changed to generally-constrained

File: pages/ecosystem/solvers.md (hunk 4 above)
Comment (abelsiqueira) [reply pages/ecosystem/solvers.md]: Good idea. Done.

File: pages/ecosystem/solvers.md (hunk 6 above)
Comment (abelsiqueira) [reply pages/ecosystem/solvers.md:17]: I changed the CSS

**Review by tmigot — APPROVED** [review]

File: pages/ecosystem/solvers.md [hunk 7]
```diff
@@ -2,4 +2,44 @@
 
 # Solvers Ecosystem
 
-A solver is an algorithm designed to solve optimization problems.
\ No newline at end of file
+The solver ecosystem in JSO is in constant development.
+We are looking into unifying the solver structure to plug-and-play usage of our solvers, subsolvers, and tools.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:6]:
```suggestion
We are looking into unifying the solver structure to make solvers, subsolvers, and tools plug-and-play.
```

File: pages/ecosystem/solvers.md [hunk 8]
```diff
@@ -2,4 +2,44 @@
...
+
+Within JSO, our solver development has focused on three types of packages:
+- New methods, usually accompanied by research papers;
+- Our versions of well-known methods;
```
Comment (dpo) [inline pages/ecosystem/solvers.md:10]:
```suggestion
- our implementation of (well- or no-so-well-)known methods;
```

File: pages/ecosystem/solvers.md [hunk 9]
```diff
@@ -2,4 +2,44 @@
...
+- Wrappers around noteworthy solvers.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:11]:
```suggestion
- wrappers around noteworthy solvers implemented in other languages.
```

File: pages/ecosystem/solvers.md [hunk 10]
```diff
@@ -2,4 +2,44 @@
...
+
+## Solver list
+
+Below you can see the list of solvers and the type of problems solved by them.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:15]:
```suggestion
Below is the list of solvers and the type of problems that they are designed to solve.
```

File: pages/ecosystem/solvers.md [hunk 11]
```diff
@@ -2,4 +2,44 @@
...
+
+| Solver     | Package       | Description                                                  | Types of problem         |
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
+| `lbfgs`    | JSOSolvers.jl | Factorization-free line search limited memory inverse BFGS.  | Unconstrained            |
```
Comment (dpo) [inline pages/ecosystem/solvers.md:21]:
```suggestion
| `lbfgs`    | JSOSolvers.jl | Factorization-free linesearch limited-memory inverse BFGS.   | Unconstrained            |
```

File: pages/ecosystem/solvers.md [hunk 12]
```diff
@@ -2,4 +2,44 @@
...
+- Our versions of well-known methods;
+- Wrappers around noteworthy solvers.
+
+## Solver list
+
+Below you can see the list of solvers and the type of problems solved by them.
+
+| Solver     | Package       | Description                                                  | Types of problem         |
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
```
Comment (dpo) [inline pages/ecosystem/solvers.md:19]: no full stop in this column

File: pages/ecosystem/solvers.md [hunk 13]
```diff
@@ -2,4 +2,44 @@
...
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
+| `lbfgs`    | JSOSolvers.jl | Factorization-free line search limited memory inverse BFGS.  | Unconstrained            |
+| `percival` | Percival.jl   | Factorization-free augmented Lagrangian.                     | Generally-constrained    |
+| `ripqp`    | RipQP.jl      | Regularized Interior-Point Quadratic Programming.            | Linearly-constrained QP  |
```
Comment (dpo) [inline pages/ecosystem/solvers.md:23]:
```suggestion
| `ripqp`    | RipQP.jl      | Regularized Interior-Point Quadratic Programming.            | Convex QP (incl. LP)  |
```

File: pages/ecosystem/solvers.md [hunk 14]
```diff
@@ -2,4 +2,44 @@
...
+- Wrappers around noteworthy solvers.
+
+## Solver list
+
+Below you can see the list of solvers and the type of problems solved by them.
+
+| Solver     | Package       | Description                                                  | Types of problem         |
+| ---------- | ------------- | ------------------------------------------------------------ | ------------------------ |
+| `cannoles` | CaNNOLeS.jl   | Regularization method for nonlinear least squares.           | Equality-constrained NLS |
+| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained     |
```
Comment (dpo) [inline pages/ecosystem/solvers.md:20]:
```suggestion
| `dci`      | DCISolver.jl  | Trust-cylinder similar to two-step SQP.                      | Equality-constrained NLP    |
```
(and similarly in other rows)

File: pages/ecosystem/solvers.md [hunk 15]
```diff
@@ -2,4 +2,44 @@
...
+| `lbfgs`    | JSOSolvers.jl | Factorization-free line search limited memory inverse BFGS.  | Unconstrained            |
+| `percival` | Percival.jl   | Factorization-free augmented Lagrangian.                     | Generally-constrained    |
+| `ripqp`    | RipQP.jl      | Regularized Interior-Point Quadratic Programming.            | Linearly-constrained QP  |
+| `tron`     | JSOSolvers.jl | Factorization-free trust-region second-order method.         | Bound-constrained        |
```
Comment (dpo) [inline pages/ecosystem/solvers.md:24]: 2nd-order or quasi-Newton (depending on the model).

File: pages/ecosystem/solvers.md [hunk 16]
```diff
@@ -2,4 +2,44 @@
...
+| `tronls`   | JSOSolvers.jl | Least-squares versions of `tron`.                            | Unconstrained NLS        |
+| `trunk`    | JSOSolvers.jl | Factorization-free trust-region second-order method.         | Unconstrained            |
+| `trunkls`  | JSOSolvers.jl | Least-squares version of `trunk`.                            | Unconstrained NLS        |
+
+## Types of problem
```
Comment (dpo) [inline pages/ecosystem/solvers.md:29]:
```suggestion
## Types of problems
```

File: pages/ecosystem/solvers.md [hunk 17]
```diff
@@ -2,4 +2,44 @@
...
+
+Our optimization problems can be defined as
+
+$$\text{minimize} \quad f(x) \quad \text{subject to} \quad x \in \Omega.$$
+
+Our constraint types and their meanings are:
+- Unconstrained: $\Omega = \R^n$.
+- Bound-constrained: $\Omega = \{x \in \R^n |\ \ell \leq x \leq u\}$.
+- Equality-constraints: $\Omega = \{x \in \R^n |\ c(x) = 0\}$.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:38]:
```suggestion
- Equality constrained: $\Omega = \{x \in \R^n |\ c(x) = 0\}$.
```

File: pages/ecosystem/solvers.md [hunk 18]
```diff
@@ -2,4 +2,44 @@
...
+
+## Types of problem
+
+Our optimization problems can be defined as
+
+$$\text{minimize} \quad f(x) \quad \text{subject to} \quad x \in \Omega.$$
+
+Our constraint types and their meanings are:
+- Unconstrained: $\Omega = \R^n$.
+- Bound-constrained: $\Omega = \{x \in \R^n |\ \ell \leq x \leq u\}$.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:37]:
```suggestion
- Bound constrained: $\Omega = \{x \in \R^n |\ \ell \leq x \leq u\}$.
```

File: pages/ecosystem/solvers.md [hunk 19]
```diff
@@ -2,4 +2,44 @@
...
+- Equality-constraints: $\Omega = \{x \in \R^n |\ c(x) = 0\}$.
+- Any constrained: $\Omega = \{x \in \R^n |\ c_L \leq c(x) \leq c_U \}$, where $c_{L_i} = c_{U_i}$ is a valid possibility, as are $c_{L_i} = -\infty$ and $c_{U_i} = \infty$, but not at the same time.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:39]:
```suggestion
- Generally constrained: $\Omega = \{x \in \R^n |\ c_L \leq c(x) \leq c_U \}$, where $c_{L_i} = c_{U_i}$ is a valid possibility, as are $c_{L_i} = -\infty$ and $c_{U_i} = \infty$, but not at the same time.
```

File: pages/ecosystem/solvers.md [hunk 20]
```diff
@@ -2,4 +2,44 @@
...
+
+In addition to constraints differences, we also have different objective types:
```
Comment (dpo) [inline pages/ecosystem/solvers.md:41]:
```suggestion
In addition to differences in the constraints, we also have different objective types:
```

File: pages/ecosystem/solvers.md [hunk 21]
```diff
@@ -2,4 +2,44 @@
...
+- **NLS**: Nonlinear Least Squares problems.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:42]:
```suggestion
- **NLS**: Nonlinear Least-Squares problems.
```

File: pages/ecosystem/solvers.md [hunk 22]
```diff
@@ -2,4 +2,46 @@
...
+
+Our constraint types and their meanings are:
+- Unconstrained: $\Omega = \R^n$.
+- Bound constrained: $\Omega = \{x \in \R^n |\ \ell \leq x \leq u\}$.
+- Equality constrained: $\Omega = \{x \in \R^n |\ c(x) = 0\}$.
+- Generally constrained: $\Omega = \{x \in \R^n |\ c_L \leq c(x) \leq c_U \}$, where $c_{L_i} = c_{U_i}$ is a valid possibility, as are $c_{L_i} = -\infty$ and $c_{U_i} = \infty$, but not at the same time.
+
+In addition to differences in the constraints, we also have different objective types:
+- **NLP**: General objective.
+  The objective $f(x)$ of these problems have no special structure being explored. Access its derivatives through the default NLPModels.jl API.
```
Comment (dpo) [inline pages/ecosystem/solvers.md:43]:
```suggestion
  The objective $f(x)$ of these problems has no special structure to exploit. Access to its derivatives occurs through the default NLPModels.jl API.
```

**Comment by dpo** [comment]
Thanks! It all looks good to me after that last comment.

**Comment by abelsiqueira** [comment]
Thanks!

**Pull request merged**